distributed/tests/test_client.py

```diff
@@ -3532,9 +3532,6 @@ def test_reconnect(loop):
         assert time() < start + 5
         sleep(0.01)
 
-    with pytest.raises(Exception):
```

Comment:
Unfortunately, this test failed, and this change made it work again. What seemed to happen is that now the client detects connections issues earlier than it used to (yay), and tries *here* to reconnect to the scheduler. This can never succeed, as there is no scheduler process, so the client eventually gives up completely (c.state is "closed").
I think it only used to work this way because the client tried to use the (closed) scheduler conneciton, triggering an exception here instead of re-trying to reconnect until the timeout is reached.

 I think the new behavior is fine.